appservice/src/deploy/deployJar.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebSiteManagementModels } from '@azure/arm-appservice';
+import { createKuduClient } from '../createKuduClient';
+import { SiteClient } from '../SiteClient';
+import { delayFirstWebAppDeploy } from './delayFirstWebAppDeploy';
+import { IDeployContext } from './IDeployContext';
+import { Readable } from 'stream';
+import * as yazl from 'yazl';
+import { waitForDeploymentToComplete } from './waitForDeploymentToComplete';
+
+export async function deployJar(context: IDeployContext, client: SiteClient, fsPath: string, aspPromise: Promise<WebSiteManagementModels.AppServicePlan | undefined>): Promise<void> {
```

Comment:
I would prefer if you used the solution I suggested in this comment https://github.com/microsoft/vscode-azuretools/pull/912#discussion_r635337289.

This solution results in a lot of duplicated code.

Reply:
Thanks for your advices, pr updated